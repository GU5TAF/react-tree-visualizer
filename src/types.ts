import React, { ReactElement } from 'react';

export type Theme = {
  fontFamily: string;

  transitionTimingFunction: string;

  disabledColor: string;

  matchBackground: {
    wonColor: string;
    lostColor: string;
  };

  border: {
    color: string;
    highlightedColor: string;
  };

  textColor: {
    highlighted: string;
    main: string;
    dark: string;
    disabled: string;
  };

  score: {
    text: {
      highlightedWonColor: string;
      highlightedLostColor: string;
    };
    background: {
      wonColor: string;
      lostColor: string;
    };
  };

  canvasBackground: string;
};

export type Options = {
  width?: number;

  boxHeight?: number;

  canvasPadding?: number;

  spaceBetweenColumns?: number;

  spaceBetweenRows?: number;

  connectorColor?: string;

  connectorColorHighlight?: string;

  roundHeader?: {
    isShown?: boolean;
    height?: number;
    marginBottom?: number;
    fontSize?: number;
    fontColor?: string;
    backgroundColor?: string;
    fontFamily?: string;
  };

  roundSeparatorWidth?: number;

  lineInfo?: {
    separation?: number;
    homeVisitorSpread?: number;
  };

  horizontalOffset?: number;

  wonBywalkOverText?: string;

  lostByNoShowText?: string;
};

export type ComputedOptions = Options & {
  rowHeight?: number;

  columnWidth?: number;
};

export type CommonTreeProps = {
  svgWrapper?: (props: {
    bracketWidth: number;
    bracketHeight: number;
    startAt: number[];
    children: ReactElement;
  }) => React.ReactElement;

  theme?: Theme;

  options?: { style: Options };
};

export type RenderItem<T> = {
  y: number;
  id: number;
  path: number[];
  parentId?: number;
  data: T;
};

export type RenderMatrix<T> = RenderItem<T>[][];

export type Tree<T> = { data: T; children: Tree<T>[] };

export type ItemComponentProps<T> = {
  item: T;
  onMouseEnter: () => void;
  onMouseLeave: () => void;
  isCollapsed: boolean;
  toggleCollapse: () => void;
  isHighlighted: boolean;
  connectorColor: string;
  computedStyles: ComputedOptions;
};

export type GeneralTreeProps<T> = CommonTreeProps & {
  itemComponent: (props: ItemComponentProps<T>) => JSX.Element;
  currentRound?: string;
  tree: Tree<T>;
};
