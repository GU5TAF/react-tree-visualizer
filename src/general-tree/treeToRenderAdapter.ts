import { numArrayComparison } from '../utils/numbers';
import { RenderMatrix, Tree } from '../types';

/**
 * Return data should look like this
 *  [
 *  [ 1st Column ]
 *  [ 2nd Column ]
 *  [ 3rd Column ]
 *  [ 4th Column ]
 * ]
 *
 */
export function generateRenderTree<T>(
  generalTree: Tree<T>,
  collapsedPaths: number[][] = [],
  depth: number = 0,
  verticalStartingPoint = 0,
  parentId: number | undefined = undefined,
  id: number = 0,
  path: number[] = [],
  result: RenderMatrix<T> = [[]]
): {
  verticalStartingPoint: number;
  id: number;
  newResult: RenderMatrix<T>;
} {
  let currentVerticalStartingPoint = verticalStartingPoint;
  let currentResult = result;
  let currentIterationId = id;

  const { children, data, ...rest } = generalTree;
  const item = {
    ...rest,
    id: currentIterationId,
    parentId,
    y: currentVerticalStartingPoint,
    path,
    data,
  };

  currentResult[depth] = currentResult[depth]?.length
    ? [...currentResult[depth], item]
    : [item];

  if (
    children.length === 0 ||
    collapsedPaths.some(collapsedPath =>
      numArrayComparison(collapsedPath, path)
    )
  ) {
    return {
      verticalStartingPoint: currentVerticalStartingPoint + 1,
      newResult: currentResult,
      id: currentIterationId,
    };
  }

  for (let i = 0; i < children.length; i++) {
    const {
      newResult,
      verticalStartingPoint: resultingVerticalStartingPoint,
      id: resultId,
    } = generateRenderTree(
      children[i],
      collapsedPaths,
      depth + 1,
      currentVerticalStartingPoint,
      id,
      currentIterationId + 1,
      [...path, i],
      currentResult
    );
    currentIterationId = resultId;
    currentVerticalStartingPoint = resultingVerticalStartingPoint;
    currentResult = newResult;
  }

  return {
    verticalStartingPoint: currentVerticalStartingPoint,
    newResult: currentResult,
    id: currentIterationId,
  };
}
