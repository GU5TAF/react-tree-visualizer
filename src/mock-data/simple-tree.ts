import { DebugItemDataType, Tree } from '../types';

export const tinyTree: Tree<DebugItemDataType> = {
  id: '0',
  data: { type: 0, id: 0 },
  children: [
    {
      id: '1',
      data: {
        id: 1,
        disabled: false,
        hidden: false,

        type: 1,
      },
      children: [
        {
          id: '2',
          children: [],
          data: {
            disabled: false,
            hidden: false,
            id: 2,

            type: 1,
          },
        },
        {
          id: '3',
          children: [],
          data: {
            id: 3,
            disabled: false,
            hidden: false,
            type: 2,
          },
        },
      ],
    },
    {
      id: '4',
      data: {
        disabled: false,
        hidden: false,

        type: 2,
        id: 4,
      },
      children: [
        {
          id: '5',
          children: [],
          data: {
            disabled: false,
            hidden: false,
            id: 5,

            type: 1,
          },
        },
      ],
    },
  ],
};

export const simpleTree: Tree<DebugItemDataType> = {
  id: '0',
  data: { id: 0, type: 0 },
  children: [
    {
      id: '4',
      data: {
        type: 1,
        id: 4,

        disabled: false,
        hidden: false,
      },
      children: [
        {
          id: '14',
          data: {
            type: 1,
            id: 14,
            disabled: false,
            hidden: false,
          },
          children: [{ id: '45', data: { type: 0, id: 45 }, children: [] }],
        },
        {
          id: '15',
          data: {
            type: 1,
            id: 15,
            disabled: false,
            hidden: false,
          },
          children: [
            {
              id: '17',
              data: {
                scrollAxis: 1,
                type: 1,
                id: 17,
                disabled: false,
                hidden: false,
              },
              children: [
                {
                  id: '22',
                  data: {
                    type: 0,
                    id: 22,
                  },
                  children: [],
                },
                {
                  id: '23',
                  data: { type: 0, id: 23 },
                  children: [],
                },
                {
                  id: '24',
                  data: { type: 0, id: 24 },
                  children: [],
                },
                {
                  id: '25',
                  data: { type: 0, id: 25 },
                  children: [],
                },
                {
                  id: '26',
                  data: { type: 0, id: 26 },
                  children: [],
                },
                {
                  id: '27',
                  data: { type: 0, id: 27 },
                  children: [],
                },
                {
                  id: '28',
                  data: { type: 0, id: 28 },
                  children: [],
                },
                {
                  id: '29',
                  data: { type: 0, id: 29 },
                  children: [],
                },
              ],
            },
            {
              id: '19',
              data: {
                scrollAxis: 1,
                type: 1,
                id: 19,
                disabled: false,
                hidden: false,
              },
              children: [
                {
                  id: '21',
                  data: { type: 0, id: 21 },
                  children: [],
                },
                {
                  id: '20',
                  data: {
                    type: 0,
                    id: 20,
                    disabled: true,
                  },
                  children: [],
                },
                {
                  id: '30',
                  data: {
                    type: 0,
                    id: 30,
                  },
                  children: [],
                },
                {
                  id: '31',
                  data: {
                    type: 0,
                    id: 31,
                  },
                  children: [],
                },
                {
                  id: '32',
                  data: {
                    type: 0,
                    id: 32,
                    disabled: true,
                  },
                  children: [],
                },
                {
                  id: '33',
                  data: {
                    type: 0,
                    id: 33,
                    disabled: true,
                  },
                  children: [],
                },
                {
                  id: '34',
                  data: {
                    type: 0,
                    id: 34,
                    disabled: true,
                  },
                  children: [],
                },
                {
                  id: '35',
                  data: { type: 0, id: 35 },
                  children: [],
                },
                {
                  id: '36',
                  data: { type: 0, id: 36 },
                  children: [],
                },
                {
                  id: '37',
                  data: {
                    type: 0,
                    id: 37,
                  },
                  children: [],
                },
                {
                  id: '38',
                  data: { type: 0, id: 38 },
                  children: [],
                },
                {
                  id: '39',
                  data: { type: 0, id: 39 },
                  children: [],
                },
                {
                  id: '40',
                  data: { type: 0, id: 40 },
                  children: [],
                },
                {
                  id: '41',
                  data: {
                    type: 0,
                    id: 41,
                  },
                  children: [],
                },
                {
                  id: '42',
                  data: { type: 0, id: 42 },
                  children: [],
                },
                {
                  id: '43',
                  data: { type: 0, id: 43 },
                  children: [],
                },
                {
                  id: '44',
                  data: {
                    type: 0,
                    id: 44,
                  },
                  children: [],
                },
              ],
            },
          ],
        },
      ],
    },
  ],
};
export const simpleTreeMutatedAtLeaf: Tree<DebugItemDataType> = {
  id: '0',
  data: { id: 0, type: 0 },
  children: [
    {
      id: '4',
      data: {
        type: 1,
        id: 4,

        disabled: false,
        hidden: false,
      },
      children: [
        {
          id: '14',
          data: {
            type: 1,
            id: 14,
            disabled: false,
            hidden: false,
          },
          children: [{ id: '45', data: { type: 0, id: 45 }, children: [] }],
        },
        {
          id: '15',
          data: {
            type: 1,
            id: 15,
            disabled: false,
            hidden: false,
          },
          children: [
            {
              id: '17',
              data: {
                scrollAxis: 1,
                type: 1,
                id: 17,
                disabled: false,
                hidden: false,
              },
              children: [
                {
                  id: '22',
                  data: {
                    type: 0,
                    id: 22,
                  },
                  children: [],
                },
                {
                  id: '23',
                  data: { type: 0, id: 23 },
                  children: [],
                },
                {
                  id: '24',
                  data: { type: 0, id: 24 },
                  children: [],
                },
                {
                  id: '25',
                  data: { type: 0, id: 25 },
                  children: [],
                },
                {
                  id: '26',
                  data: { type: 0, id: 26 },
                  children: [],
                },
                {
                  id: '27',
                  data: { type: 0, id: 27 },
                  children: [],
                },
                {
                  id: '28',
                  data: { type: 0, id: 28 },
                  children: [],
                },
                {
                  id: '29',
                  data: { type: 0, id: 29 },
                  children: [],
                },
              ],
            },
            {
              id: '19',
              data: {
                scrollAxis: 1,
                type: 1,
                id: 19,
                disabled: false,
                hidden: false,
              },
              children: [
                {
                  id: '30',
                  data: {
                    type: 0,
                    id: 30,
                  },
                  children: [],
                },
                {
                  id: '31',
                  data: {
                    type: 0,
                    id: 31,
                  },
                  children: [],
                },
                {
                  id: '32',
                  data: {
                    type: 0,
                    id: 32,
                    disabled: true,
                  },
                  children: [],
                },
                {
                  id: '33',
                  data: {
                    type: 0,
                    id: 33,
                    disabled: true,
                  },
                  children: [],
                },
                {
                  id: '34',
                  data: {
                    type: 0,
                    id: 34,
                    disabled: true,
                  },
                  children: [],
                },
                {
                  id: '35',
                  data: { type: 0, id: 35 },
                  children: [],
                },
                {
                  id: '36',
                  data: { type: 0, id: 36 },
                  children: [],
                },
                {
                  id: '37',
                  data: {
                    type: 0,
                    id: 37,
                  },
                  children: [],
                },
                {
                  id: '38',
                  data: { type: 0, id: 38 },
                  children: [],
                },
                {
                  id: '39',
                  data: { type: 0, id: 39 },
                  children: [],
                },
                {
                  id: '40',
                  data: { type: 0, id: 40 },
                  children: [],
                },
                {
                  id: '41',
                  data: {
                    type: 0,
                    id: 41,
                  },
                  children: [],
                },
                {
                  id: '42',
                  data: { type: 0, id: 42 },
                  children: [],
                },
                {
                  id: '43',
                  data: { type: 0, id: 43 },
                  children: [],
                },
                {
                  id: '44',
                  data: {
                    type: 0,
                    id: 44,
                  },
                  children: [],
                },
              ],
            },
          ],
        },
      ],
    },
  ],
};

export const simpleTreeMutatedAtBranch: Tree<DebugItemDataType> = {
  id: '0',
  data: { id: 0, type: 0 },
  children: [
    {
      id: '4',
      data: {
        type: 1,
        id: 4,

        disabled: false,
        hidden: false,
      },
      children: [
        {
          id: '14',
          data: {
            type: 1,
            id: 14,
            disabled: false,
            hidden: false,
          },
          children: [{ id: '45', data: { type: 0, id: 45 }, children: [] }],
        },
        {
          id: '15',
          data: {
            type: 1,
            id: 15,
            disabled: false,
            hidden: false,
          },
          children: [
            {
              id: '19',
              data: {
                scrollAxis: 1,
                type: 1,
                id: 19,
                disabled: false,
                hidden: false,
              },
              children: [
                {
                  id: '21',
                  data: { type: 0, id: 21 },
                  children: [],
                },
                {
                  id: '20',
                  data: {
                    type: 0,
                    id: 20,
                    disabled: true,
                  },
                  children: [],
                },
                {
                  id: '30',
                  data: {
                    type: 0,
                    id: 30,
                  },
                  children: [],
                },
                {
                  id: '31',
                  data: {
                    type: 0,
                    id: 31,
                  },
                  children: [],
                },
                {
                  id: '32',
                  data: {
                    type: 0,
                    id: 32,
                    disabled: true,
                  },
                  children: [],
                },
                {
                  id: '33',
                  data: {
                    type: 0,
                    id: 33,
                    disabled: true,
                  },
                  children: [],
                },
                {
                  id: '34',
                  data: {
                    type: 0,
                    id: 34,
                    disabled: true,
                  },
                  children: [],
                },
                {
                  id: '35',
                  data: { type: 0, id: 35 },
                  children: [],
                },
                {
                  id: '36',
                  data: { type: 0, id: 36 },
                  children: [],
                },
                {
                  id: '37',
                  data: {
                    type: 0,
                    id: 37,
                  },
                  children: [],
                },
                {
                  id: '38',
                  data: { type: 0, id: 38 },
                  children: [],
                },
                {
                  id: '39',
                  data: { type: 0, id: 39 },
                  children: [],
                },
                {
                  id: '40',
                  data: { type: 0, id: 40 },
                  children: [],
                },
                {
                  id: '41',
                  data: {
                    type: 0,
                    id: 41,
                  },
                  children: [],
                },
                {
                  id: '42',
                  data: { type: 0, id: 42 },
                  children: [],
                },
                {
                  id: '43',
                  data: { type: 0, id: 43 },
                  children: [],
                },
                {
                  id: '44',
                  data: {
                    type: 0,
                    id: 44,
                  },
                  children: [],
                },
              ],
            },
          ],
        },
      ],
    },
  ],
};
