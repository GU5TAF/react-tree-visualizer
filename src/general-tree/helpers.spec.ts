import { DebugItemDataType, RenderMatrix } from 'types';
import {
  findItemFromRenderMatrix,
  findPathFromItem,
  processTree,
} from './helpers';
import { simpleTree, tinyTree } from '../mock-data/simple-tree';
import { generateRenderTree } from './treeToRenderAdapter';

describe('findPathFromItem', () => {
  it('converts it to render data', () => {
    const renderMatrix = processTree(simpleTree);
    const columnIndex = 3;
    const rowIndex = 4;
    const testItem = renderMatrix[columnIndex][rowIndex];

    const result = findPathFromItem<DebugItemDataType>(
      [testItem.uniqueId],
      testItem.parentUniqueId,
      renderMatrix.slice(0, Math.max(columnIndex, 0))
    );

    expect(result).toMatchSnapshot();
  });
});

describe('findItemFromRenderMatrix', () => {
  it('converts it to render data', () => {
    const renderMatrix = processTree(simpleTree);
    const columnIndex = 3;
    const rowIndex = 4;
    const testItem = renderMatrix[columnIndex][rowIndex];

    const { item: resultItem } = findItemFromRenderMatrix<DebugItemDataType>(
      renderMatrix,
      testItem.uniqueId
    );

    expect(resultItem).toMatchObject(testItem);
  });
});

// const renderMatrix: RenderMatrix<DebugItemDataType> = [
//   [
//     {
//       uniqueId: '4',
//       id: 1,
//       parentUniqueId: '0',
//       parentId: 0,
//       y: 0,
//       path: [0],
//       data: { type: 1, id: 4, disabled: false, hidden: false },
//     },
//   ],
//   [
//     {
//       uniqueId: '14',
//       id: 2,
//       parentUniqueId: '4',
//       parentId: 1,
//       y: 0,
//       path: [0, 0],
//       data: { type: 1, id: 14, disabled: false, hidden: false },
//     },
//     {
//       uniqueId: '15',
//       id: 4,
//       parentUniqueId: '4',
//       parentId: 1,
//       y: 1,
//       path: [0, 1],
//       data: { type: 1, id: 15, disabled: false, hidden: false },
//     },
//   ],
//   [
//     {
//       uniqueId: '45',
//       id: 3,
//       parentUniqueId: '14',
//       parentId: 2,
//       y: 0,
//       path: [0, 0, 0],
//       data: { type: 0, id: 45 },
//     },
//     {
//       uniqueId: '17',
//       id: 5,
//       parentUniqueId: '15',
//       parentId: 4,
//       y: 1,
//       path: [0, 1, 0],
//       data: { scrollAxis: 1, type: 1, id: 17, disabled: false, hidden: false },
//     },
//     {
//       uniqueId: '19',
//       id: 14,
//       parentUniqueId: '15',
//       parentId: 4,
//       y: 9,
//       path: [0, 1, 1],
//       data: { scrollAxis: 1, type: 1, id: 19, disabled: false, hidden: false },
//     },
//   ],
//   [
//     {
//       uniqueId: '22',
//       id: 6,
//       parentUniqueId: '17',
//       parentId: 5,
//       y: 1,
//       path: [0, 1, 0, 0],
//       data: { type: 0, id: 22 },
//     },
//     {
//       uniqueId: '23',
//       id: 7,
//       parentUniqueId: '17',
//       parentId: 5,
//       y: 2,
//       path: [0, 1, 0, 1],
//       data: { type: 0, id: 23 },
//     },
//     {
//       uniqueId: '24',
//       id: 8,
//       parentUniqueId: '17',
//       parentId: 5,
//       y: 3,
//       path: [0, 1, 0, 2],
//       data: { type: 0, id: 24 },
//     },
//     {
//       uniqueId: '25',
//       id: 9,
//       parentUniqueId: '17',
//       parentId: 5,
//       y: 4,
//       path: [0, 1, 0, 3],
//       data: { type: 0, id: 25 },
//     },
//     {
//       uniqueId: '26',
//       id: 10,
//       parentUniqueId: '17',
//       parentId: 5,
//       y: 5,
//       path: [0, 1, 0, 4],
//       data: { type: 0, id: 26 },
//     },
//     {
//       uniqueId: '27',
//       id: 11,
//       parentUniqueId: '17',
//       parentId: 5,
//       y: 6,
//       path: [0, 1, 0, 5],
//       data: { type: 0, id: 27 },
//     },
//     {
//       uniqueId: '28',
//       id: 12,
//       parentUniqueId: '17',
//       parentId: 5,
//       y: 7,
//       path: [0, 1, 0, 6],
//       data: { type: 0, id: 28 },
//     },
//     {
//       uniqueId: '29',
//       id: 13,
//       parentUniqueId: '17',
//       parentId: 5,
//       y: 8,
//       path: [0, 1, 0, 7],
//       data: { type: 0, id: 29 },
//     },
//     {
//       uniqueId: '21',
//       id: 15,
//       parentUniqueId: '19',
//       parentId: 14,
//       y: 9,
//       path: [0, 1, 1, 0],
//       data: { type: 0, id: 21 },
//     },
//     {
//       uniqueId: '20',
//       id: 16,
//       parentUniqueId: '19',
//       parentId: 14,
//       y: 10,
//       path: [0, 1, 1, 1],
//       data: { type: 0, id: 20, disabled: true },
//     },
//     {
//       uniqueId: '30',
//       id: 17,
//       parentUniqueId: '19',
//       parentId: 14,
//       y: 11,
//       path: [0, 1, 1, 2],
//       data: { type: 0, id: 30 },
//     },
//     {
//       uniqueId: '31',
//       id: 18,
//       parentUniqueId: '19',
//       parentId: 14,
//       y: 12,
//       path: [0, 1, 1, 3],
//       data: { type: 0, id: 31 },
//     },
//     {
//       uniqueId: '32',
//       id: 19,
//       parentUniqueId: '19',
//       parentId: 14,
//       y: 13,
//       path: [0, 1, 1, 4],
//       data: { type: 0, id: 32, disabled: true },
//     },
//     {
//       uniqueId: '33',
//       id: 20,
//       parentUniqueId: '19',
//       parentId: 14,
//       y: 14,
//       path: [0, 1, 1, 5],
//       data: { type: 0, id: 33, disabled: true },
//     },
//     {
//       uniqueId: '34',
//       id: 21,
//       parentUniqueId: '19',
//       parentId: 14,
//       y: 15,
//       path: [0, 1, 1, 6],
//       data: { type: 0, id: 34, disabled: true },
//     },
//     {
//       uniqueId: '35',
//       id: 22,
//       parentUniqueId: '19',
//       parentId: 14,
//       y: 16,
//       path: [0, 1, 1, 7],
//       data: { type: 0, id: 35 },
//     },
//     {
//       uniqueId: '36',
//       id: 23,
//       parentUniqueId: '19',
//       parentId: 14,
//       y: 17,
//       path: [0, 1, 1, 8],
//       data: { type: 0, id: 36 },
//     },
//     {
//       uniqueId: '37',
//       id: 24,
//       parentUniqueId: '19',
//       parentId: 14,
//       y: 18,
//       path: [0, 1, 1, 9],
//       data: { type: 0, id: 37 },
//     },
//     {
//       uniqueId: '38',
//       id: 25,
//       parentUniqueId: '19',
//       parentId: 14,
//       y: 19,
//       path: [0, 1, 1, 10],
//       data: { type: 0, id: 38 },
//     },
//     {
//       uniqueId: '39',
//       id: 26,
//       parentUniqueId: '19',
//       parentId: 14,
//       y: 20,
//       path: [0, 1, 1, 11],
//       data: { type: 0, id: 39 },
//     },
//     {
//       uniqueId: '40',
//       id: 27,
//       parentUniqueId: '19',
//       parentId: 14,
//       y: 21,
//       path: [0, 1, 1, 12],
//       data: { type: 0, id: 40 },
//     },
//     {
//       uniqueId: '41',
//       id: 28,
//       parentUniqueId: '19',
//       parentId: 14,
//       y: 22,
//       path: [0, 1, 1, 13],
//       data: { type: 0, id: 41 },
//     },
//     {
//       uniqueId: '42',
//       id: 29,
//       parentUniqueId: '19',
//       parentId: 14,
//       y: 23,
//       path: [0, 1, 1, 14],
//       data: { type: 0, id: 42 },
//     },
//     {
//       uniqueId: '43',
//       id: 30,
//       parentUniqueId: '19',
//       parentId: 14,
//       y: 24,
//       path: [0, 1, 1, 15],
//       data: { type: 0, id: 43 },
//     },
//     {
//       uniqueId: '44',
//       id: 31,
//       parentUniqueId: '19',
//       parentId: 14,
//       y: 25,
//       path: [0, 1, 1, 16],
//       data: { type: 0, id: 44 },
//     },
//   ],
// ];
