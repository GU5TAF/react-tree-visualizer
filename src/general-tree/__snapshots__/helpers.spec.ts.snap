// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`findPathFromItem converts it to render data 1`] = `
Array [
  "4",
  "15",
  "17",
  "26",
]
`;
